CONTRIBUTING.md

```diff
@@ -0,0 +1,355 @@
+   - Make sure your new tests cover all different edge cases.
+   - Make sure all exceptions raised are tested.
+   - Make sure all warnings raised are tested.
+
+4. If your PR is not ready for reviews, but you want to run it on our CI, open a "Draft PR" to let us know you don't need feedback yet.
+
+5. If any of the existing tests fail in your PR on our CI, identify what's failing and try to address it.
+
+6. When you feel ready for integrating your work, mark your PR "Ready for review".
+
+   - Your code should be readable and follow the project's design principles.
+   - Make sure all tests are passing and any new code is tested for (coverage!).
+   - Make sure you link the GitHub issue to your PR.
+   - Make sure any docs for that piece of code are updated, or added.
+   - The code should be elegant and simple. No over-engineering or hard-to-read code.
+
+   Do your best but don't sweat about perfection! We do code-review to find any missed items.
+   If you need help, don't hesitate to ping the core team on the PR.
+
+7. Use tags in PR name for the following cases:
+
+   - **\[blocked by #<number>\]** if your work is dependent on other PRs.
+   - **\[wip\]** when you start to re-edit your work, mark it so no one will accidentally merge it in meantime.
+
+______________________________________________________________________
+
+### Setup LitData
+
+We use Makefile for easier setup. To setup LitData, follow these steps:
+
+1. Clone the forked-repository
+
+```bash
+git clone https://github.com/{YOUR_USERNAME}/litdata.git
+cd litdata
+```
+
+2. Optional, but recommended: Create a virtual conda environment. If you're using [Lightning Studio](https://lightning.ai/studios), skip this step.
+
+```bash
+conda create -n litdata python=3.10
+conda activate litdata
+```
+
+3. Make sure you have `make` installed
+
+```bash
+# for debian-based systems: sudo apt-get install build-essential
+# for mac: brew install make
+make --version
+```
+
+4. Run command:
+
+```bash
+make setup
+```
+
+That's it. You are ready to go! 🎉
+
+______________________________________________________________________
+
+### Question & Answer
+
+#### How can I help/contribute?
+
+All types of contributions are welcome - reporting bugs, fixing documentation, adding test cases, solving issues, and preparing bug fixes.
+To get started with code contributions, look for issues marked with the label [good first issue](https://github.com/Lightning-AI/lightning/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) or chose something close to your domain with the label [help wanted](https://github.com/Lightning-AI/lightning/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22). Before coding, make sure that the issue description is clear and comment on the issue so that we can assign it to you (or simply self-assign if you can).
```

Comment:
```suggestion
To get started with code contributions, look for issues marked with the label [good first issue](https://github.com/Lightning-AI/litdata/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) or chose something close to your domain with the label [help wanted](https://github.com/Lightning-AI/litdata/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22). Before coding, make sure that the issue description is clear and comment on the issue so that we can assign it to you (or simply self-assign if you can).
```